src/components/MintingButton.tsx

```diff
@@ -0,0 +1,160 @@
+
+interface Props {
+  projectId: string,
+  priceWei: BigNumber
+  currencySymbol: string,
+  isConnected: boolean,
+  artistCanMint: boolean,
+  anyoneCanMint: boolean,
+  scriptAspectRatio: number
+}
+
+const multiplyBigNumberByFloat = function(x: BigNumber, y: number) {
+  return BigNumber.from(Math.floor(x.toNumber()*y))
+}
+
+const formatEtherFixed = function(priceWei: string, fractionDigits: number) {
+  let priceEther = utils.formatEther(priceWei)
+  let priceEtherFixed = parseFloat(priceEther).toFixed(fractionDigits)
+  return priceEtherFixed
+}
+
+const MintingInteraction = ({ 
+  projectId,
+  priceWei,
+  currencySymbol,
+  isConnected,
+  artistCanMint,
+  anyoneCanMint,
+  scriptAspectRatio
+  }: Props) => {
+  const windowSize = useWindowSize()
+  const [dialog, setDialog] = useState("")
+  const [mintingTokenId, setMintingTokenId] = useState<any | null>(null)
+  const [mintingPreview, setMintingPreview] = useState(false)
+  const handleMintingPreviewOpen = () => setMintingPreview(true)
+  const handleMintingPreviewClose = () => setMintingPreview(false)
+
+  const { config } = usePrepareContractWrite({
+    address: MINT_CONTRACT_ADDRESS,
+    abi: GenArt721MintABI,
+    functionName: "purchase",
+    overrides: {
+      value: priceWei
+    },
+    args: [
+      BigNumber.from(projectId)
+    ]
+  })
+  
+  let customRequest = config.request ? {
+    data: config.request?.data,
+    from: config.request?.from,
+    gasLimit: multiplyBigNumberByFloat(config.request?.gasLimit, MULTIPLY_GAS_LIMIT),
+    to: config.request?.to,
+    value: config.request?.value
+  } : undefined
+
+  const { data, isError, isLoading, write } = useContractWrite({
+    ...config,
+    request: customRequest,
+    onSuccess(data) {
+      setDialog("Transaction pending...")
+    }
+  })
+  const waitForTransaction = useWaitForTransaction({
+    hash: data?.hash,
+    confirmations: 1,
+    onSuccess(data) {
+      let tokenId = data?.logs[0]?.topics[3]
+      if (tokenId) {  
+        setMintingTokenId(parseInt(tokenId, 16).toString())
+        handleMintingPreviewOpen()
+      }
+      setDialog("")
+    }
+  })
+
+  return (
+    <>
+      {
+        artistCanMint && 
+        (
+          <MintingButtonEnabled 
+            message={`Artist Mint for ${formatEtherFixed(priceWei.toString(), 3)} ${currencySymbol}`} 
+            contractPurchase={write}
+          />
+        )
+      }
+      {
+        anyoneCanMint && 
+        (
+          <MintingButtonEnabled 
+            message={`Purchase for ${formatEtherFixed(priceWei.toString(), 3)} ${currencySymbol}`} 
+            contractPurchase={write}
+          />
+        )
+      }
+      {
+        anyoneCanMint && !isConnected &&
+        (
+          <Typography fontWeight={800} fontStyle="italic">
+            Connect to purchase...
+          </Typography>
+        )
+      }
+      <Box marginTop={1}>
+        <Typography>
+          <i>{dialog}</i>
```

Comment:
`i` tags should not be used to influence display logic and instead should be dropped in favor of CSS